tools/changelog-helper.sh

```diff
@@ -182,8 +181,8 @@ check_or_add_pr() {
     local changelog_before
     changelog_before=$(head -n "${changelog_begin_position}" ChangeLog)
     local changelog_after
-    changelog_after=$(tail -n "+$((changelog_begin_position+1))" ChangeLog)
-    (echo "$changelog_before"; echo; echo "$new_entry"; echo "$changelog_after") > ChangeLog
+    changelog_after=$(tail -n "+$((changelog_begin_position + 1))" ChangeLog)
+    (echo "$changelog_before" && echo && echo "$new_entry" && echo "$changelog_after") > ChangeLog
```

Comment:
Did the app complain here? What's the semantic difference? If one fails, the whole thing returns a non zero exit code?

Reply:
> Did the app complain here?

Didn't complain, but made a formatting mess. ;)

> If one fails, the whole thing returns a non zero exit code?

I think there should be no difference at all. First, I don't think `echo` (the shell built-in) can fail here at all. Second, with `set -u` any failure (including in subshells) should propagate and cause the whole script to exit.

```
$ bash -c 'set -eu; (true; false; true)'; echo $?
1
$ bash -c 'set -eu; (true && false && true)'; echo $?
1
```